Swift 5.4 builds are broken

There are a few issues in `Package.swift` that are causing builds to fail using Swift 5.4 in both Linux and Darwin environments.

## '-enable-experimental-distributed' not supported on Darwin ##

The `-enable-experimental-distributed` flag appears to be unsupported in the final Darwin release of Swift 5.4:

```
$ swift -version
Apple Swift version 5.4 (swiftlang-1205.0.26.9 clang-1205.0.19.55)
Target: x86_64-apple-darwin20.2.0

$ swift build
[1/1] Planning build

<unknown>:0: error: no such module '_Concurrency'
<unknown>:0: error: no such module '_Concurrency'
```

## swift-syntax release incompatible with Swift 5.4 ##

The `swift-DEVELOPMENT-SNAPSHOT-2021-03-25-a` release of `swift-syntax` selected for 5.4 and above doesn't appear to be compatible with the final release of Swift 5.4 on all platforms:

```
/swift-distributed-actors/.build/checkouts/swift-syntax/Sources/SwiftSyntax/SyntaxParser.swift:196:58: error: extra argument in call
      swiftparse_parse_string(c_parser, buf, source.utf8.count)
                                             ~~~~~~~~~~~~^~~~~
/swift-distributed-actors/.build/checkouts/swift-syntax/Sources/SwiftSyntax/SyntaxParser.swift:196:58: error: extra argument in call
      swiftparse_parse_string(c_parser, buf, source.utf8.count)
                                             ~~~~~~~~~~~~^~~~~
/swift-distributed-actors/.build/checkouts/swift-syntax/Sources/SwiftSyntax/SyntaxParser.swift:196:58: error: extra argument in call
      swiftparse_parse_string(c_parser, buf, source.utf8.count)
                                             ~~~~~~~~~~~~^~~~~
/swift-distributed-actors/.build/checkouts/swift-syntax/Sources/SwiftSyntax/SyntaxParser.swift:196:58: error: extra argument in call
      swiftparse_parse_string(c_parser, buf, source.utf8.count)
                                             ~~~~~~~~~~~~^~~~~
/swift-distributed-actors/.build/checkouts/swift-syntax/Sources/SwiftSyntax/SyntaxParser.swift:196:58: error: extra argument in call
      swiftparse_parse_string(c_parser, buf, source.utf8.count)
                                             ~~~~~~~~~~~~^~~~~
/swift-distributed-actors/.build/checkouts/swift-syntax/Sources/SwiftSyntax/RawSyntax.swift:120:45: error: value of type 'swiftparse_token_data_t' has no member 'range'
    let textSize = hasCustomText ? Int(data.range.length) : 0
                                       ~~~~ ^~~~~
...
```

--- a/Package.swift
+++ b/Package.swift
@@ -296,7 +296,11 @@ var dependencies: [Package.Dependency] = [
     .package(url: "https://github.com/apple/swift-argument-parser", .upToNextMinor(from: "0.3.2")), // not API stable, Apache v2
 ]
 
-#if swift(>=5.4)
+#if swift(>=5.5)
+dependencies.append(
+    .package(name: "SwiftSyntax", url: "https://github.com/apple/swift-syntax.git", .revision("swift-DEVELOPMENT-SNAPSHOT-2021-05-18-a"))
+)
+#elseif swift(>=5.4)
 dependencies.append(
     .package(name: "SwiftSyntax", url: "https://github.com/apple/swift-syntax.git", .exact("0.50400.0"))
 )

